pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentMessageHandlerFactory.java

```diff
@@ -58,6 +59,8 @@ public MessageHandler createHandler(Message message, NotificationContext context
         return new SegmentRefreshMessageHandler(new SegmentRefreshMessage(message), _metrics, context);
       case SegmentReloadMessage.RELOAD_SEGMENT_MSG_SUB_TYPE:
         return new SegmentReloadMessageHandler(new SegmentReloadMessage(message), _metrics, context);
+      case TableDeletionMessage.DELETE_TABLE_MSG_SUB_TYPE:
+        return new TableDeletionMessageHandler(new TableDeletionMessage(message), _metrics, context);
```

Comment:
Does this need a new TableDeletionMessage or the current message can be just cast to it?